packages/angular_devkit/schematics/tools/file-system-host.ts

```diff
@@ -5,28 +5,11 @@
  * Use of this source code is governed by an MIT-style license that can be
  * found in the LICENSE file at https://angular.io/license
  */
-import { fs } from '@angular-devkit/core/node';
-import { existsSync, readFileSync, readdirSync } from 'fs';
-import { join } from 'path';
-import { FileSystemTreeHost } from '../src';
+import { normalize, virtualFs } from '@angular-devkit/core';
+import { NodeJsSyncHost } from '@angular-devkit/core/node';
 
-export class FileSystemHost implements FileSystemTreeHost {
-  constructor(private _root: string) {}
-
-  listDirectory(path: string) {
-    return readdirSync(join(this._root, path));
-  }
-  isDirectory(path: string) {
-    return fs.isDirectory(join(this._root, path));
-  }
-  readFile(path: string) {
-    return readFileSync(join(this._root, path));
-  }
-  exists(path: string) {
-    return existsSync(this.join(this._root, path));
-  }
-
-  join(path1: string, path2: string) {
-    return join(path1, path2);
+export class FileSystemHost extends virtualFs.DelegateHost<{}> {
```

Comment:
Could this be implemented as:
```
export class FileSystemHost extends virtualFs.ScopedHost {
    constructor(dir: string) {
        super(new NodeJsSyncHost(), normalize(dir));
    }
 }
```

Seems to be the only place the `DelegateHost` class is used.

Reply:
Yes, probably. Done.